packages/hooks/src/useSet/index.ts

```diff
@@ -2,11 +2,8 @@ import { useState } from 'react';
 import useMemoizedFn from '../useMemoizedFn';
 
 function useSet<K>(initialValue?: Iterable<K>) {
-  const getInitValue = () => {
-    return initialValue === undefined ? new Set<K>() : new Set(initialValue);
-  };
-
-  const [set, setSet] = useState<Set<K>>(() => getInitValue());
+  const getInitValue = new Set(initialValue);
+  const [set, setSet] = useState<Set<K>>(getInitValue);
```

Comment:
`new Set(initialValue)` should be wrapped in function

Reply:
> `new Set(initialValue)` should be wrapped in function

Thank you for your review and feedback! I have made the necessary changes as per your suggestions and have implemented them in the code. Please review my code again.

Reply:
u should not wrap it using `useMemo` because it's calculate eagerly when deps change, that is not expected behavior. Initial state should only generate once when component mounting, and should not re-calculate if input `initialValue` change. Just pass initailizer fn to `useState` like this.

```suggestion
  const [set, setSet] = useState<Set<K>>(() => new Set(initialValue));
```